compiler/src/iree/compiler/DispatchCreation/test/fold_unit_dims.mlir

```diff
@@ -127,3 +127,66 @@ util.func public @scatter(%arg0 : tensor<4xi64>, %arg1 : tensor<4x1xi32>, %arg2
 //  CHECK-SAME:     ins(%[[ARG0]], %[[COLLAPSED]]
 //  CHECK-SAME:     outs(%[[ARG2]]
 //       CHECK:   util.return %[[SCATTER]]
+
+// -----
+
+util.func public @attention_mask_multi_m_dims(%arg0: tensor<8x4x1x128xf32>, %arg1: tensor<?x32x8x128xf32>, %arg2: tensor<?x32x8x128xf32>, %arg3: f32, %arg4: tensor<8x4x1x?x32xf32>) -> tensor<8x4x1x128xf32> {
+  %0 = tensor.empty() : tensor<8x4x1x128xf32>
+  %1 = iree_linalg_ext.attention {
+    indexing_maps = [
+      affine_map<(d0, d1, d2, d3, d4, d5, d6) -> (d0, d1, d2, d4)>,
+      affine_map<(d0, d1, d2, d3, d4, d5, d6) -> (d5, d6, d0, d4)>,
+      affine_map<(d0, d1, d2, d3, d4, d5, d6) -> (d5, d6, d0, d3)>,
+      affine_map<(d0, d1, d2, d3, d4, d5, d6) -> ()>,
+      affine_map<(d0, d1, d2, d3, d4, d5, d6) -> (d0, d1, d2, d5, d6)>,
+      affine_map<(d0, d1, d2, d3, d4, d5, d6) -> (d0, d1, d2, d3)>]}
+    ins(%arg0, %arg1, %arg2, %arg3, %arg4 : tensor<8x4x1x128xf32>, tensor<?x32x8x128xf32>, tensor<?x32x8x128xf32>, f32, tensor<8x4x1x?x32xf32>)
+    outs(%0 : tensor<8x4x1x128xf32>) {
+  ^bb0(%arg5: f32):
+    iree_linalg_ext.yield %arg5 : f32
+  } -> tensor<8x4x1x128xf32>
+  util.return %1 : tensor<8x4x1x128xf32>
+}
+// CHECK-LABEL: func public @attention_mask_multi_m_dims
+// CHECK-SAME:    %[[ARG0:[a-zA-Z0-9]+]]
+// CHECK-SAME:    %[[ARG1:[a-zA-Z0-9]+]]
+// CHECK-SAME:    %[[ARG2:[a-zA-Z0-9]+]]
+// CHECK-SAME:    %[[ARG3:[a-zA-Z0-9]+]]
+// CHECK-SAME:    %[[ARG4:[a-zA-Z0-9]+]]
+//       CHECK:   %[[COLLAPSED:.+]] = tensor.collapse_shape %[[ARG4]]
```

Comment:
You probably want to check you get the attention mask you want as well. Cause I think `d2` gets dropped in the final indexing map.